testing_and_setup/compass/utility_scripts/LIGHTparticles/animated_particles.py

```diff
@@ -86,48 +82,51 @@ def build_particle_file(fname_in, fname_outbase):
             if dim == particleTypeStatic:
                 particleData[str(v)] = cleanup(f_in.variables[v][:])
 
-        #print particleData
-
-        # set the new time (just use index for simplicity for now)
-        time = 2*t
-        #time = ''.join(f_in.variables['xtime'][t,0,:]).replace(' ','')
+        if (ignore_xtime) or ('xtime' not in f_in.variables):
+            # Set time to a simple index.
+            time = 2*t
+        else:
+            # Set decimal time similar to that in `paraview_vtk_extractor` so
+            # that the `annotate_date` macro from MPAS-Tools can be used.
+            xtime = f_in.variables['xtime'][t].tostring().decode('utf-8') \
+                        .strip()
+            date = datetime(int(xtime[0:4]), int(xtime[5:7]),
+                            int(xtime[8:10]), int(xtime[11:13]),
+                            int(xtime[14:16]), int(xtime[17:19]))
+            time = date2num(date, units='days since 0000-01-01',
+                            calendar='noleap')/365
 
         # file name
-        #f_out = fname_outbase + '_' + os.path.splitext(fname_in)[0] + '_' + str(t)
-        f_out = fname_outbase + '_' + os.path.splitext(fullpath[-1])[0] + '_' + str(t)
+        f_out = f'{fname_outbase}_{os.path.splitext(fullpath[-1])[0]}_{str(t)}'
+
         # output file
-        pointsToVTK(f_out, x, y, z, data = particleData)
+        pointsToVTK(f_out, x, y, z, data=particleData)
         # make sure the file is added to the time vector
-        g.addFile(filepath = f_out + '.vtu', sim_time = time)
-
+        g.addFile(filepath=f_out + '.vtu', sim_time=time)
 
     # save the animation file
     g.save()
 
 
 if __name__ == "__main__":
-    from optparse import OptionParser
-
     # Get command line parameters
-    parser = OptionParser()
-    parser.add_option("-f", "--file", dest="inputfilename",
-            help="file to open for appending \
-                    particle data 'particle_' extension",
-                    metavar="FILE")
-
-    parser.add_option("-o", "--output", dest="outputfilename",
-            help="output file name base for export to *.vtu \
-                    file which stores particle data",
-                    metavar="FILE")
-
-    options, args = parser.parse_args()
-
-    if not options.inputfilename:
-        parser.error("Input filename is a required input.")
-
-    if not options.outputfilename:
-        parser.error("Output filename base for vtu files is required input.")
-
-    build_particle_file(options.inputfilename, options.outputfilename)
+    ap = argparse.ArgumentParser(
+            description="Converts raw LIGHT output to VTK files for analysis \
+            in ParaView.")
+    ap.add_argument("-f", "--file", dest="inputfilename", required=True,
+                    type=str, help="file to open for appending particle data \
+                            'particle_' extension")
+    ap.add_argument("-o", "--output", dest="outputfilename", required=True,
```

Comment:
I think another cleanup would be to make `-o` not required. It's a tedious required flag. If I want to just output the VTK files to my current directory, I can pass `./animated_particles.py -f input.nc -o ''` which names the output files `_input.vtk` since this script *expects* an output directory. Perhaps we could make this not required, and if it isn't input, it just dumps the files in the same directory with the same name as the input file.